core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -529,10 +537,34 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  /**
+   * Returns whether the version of Spark that generated logs records app IDs. App IDs were added
+   * in Spark 1.1.
+   */
+  private def sparkVersionHasAppId(entry: FileStatus): Boolean = {
+    if (isLegacyLogDirectory(entry)) {
+      fs.listStatus(entry.getPath())
+        .find { status => status.getPath().getName().startsWith(SPARK_VERSION_PREFIX) }
+        .map { status =>
+          val version = status.getPath().getName().substring(SPARK_VERSION_PREFIX.length())
+          version != "1.0" && version != "1.1"
```

Comment:
Actually 1.1 doesn't have app IDs either: https://github.com/apache/spark/blob/branch-1.1/core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala#L92

It was added in 1.2: https://github.com/apache/spark/blob/branch-1.2/core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala#L99